src/pom.xml

```diff
@@ -587,6 +593,18 @@
             </dependency>
 
             <dependency>
+                <groupId>org.apache.httpcomponents</groupId>
+                <artifactId>httpclient</artifactId>
+                <version>${apache-httpclient.version}</version>
+            </dependency>
+
+            <dependency>
+                <groupId>javax.servlet</groupId>
+                <artifactId>javax.servlet-api</artifactId>
+                <version>${javax-servlet-api.version}</version>
```

Comment:
The answer is: We **can't** use `provided` here as we are explicitly using `javax.servlet.http.HttpServletRequest` outside of any servlet container (e.g. in the `util`) module.